packages/peregrine/lib/hooks/useMediaQuery.js

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from 'react';
+import { string, shape, object, arrayOf } from 'prop-types';
+
+const { matchMedia } = globalThis;
+
+/**
+ * A hook that will return all matched styles for any given media queries using the
+ * matchMedia API (https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia)
+ *
+ * @param {Object} props
+ * @param {Array} props.mediaQueries The array of media rules and respective styles to apply
+ *
+ * @returns {Object}
+ */
+export const useMediaQuery = (props = { mediaQueries: [] }) => {
+    const [styles, setStyles] = useState({});
+
+    const { mediaQueries } = props;
+
+    useEffect(() => {
+        if (!mediaQueries) return;
+
+        const mqlList = mediaQueries.map(({ media }) => matchMedia(media));
+
+        const handleMatch = (query, i) => {
```

Comment:
can we add unit tests that cover both cases in this logic?

Reply:
Ok, I can try mocking unmatched queries.